packages/plugin-hardhat/src/admin.ts

```diff
@@ -59,14 +62,16 @@ export function makeTransferProxyAdminOwnership(
     const overrides = opts.txOverrides ? [opts.txOverrides] : [];
     await admin.transferOwnership(newOwner, ...overrides);
 
-    const { provider } = hre.network;
-    const manifest = await Manifest.forNetwork(provider);
-    const { proxies } = await manifest.read();
-    for (const { address, kind } of proxies) {
-      if ((await admin.getAddress()) == (await getAdminAddress(provider, address))) {
-        console.log(SUCCESS_CHECK + `${address} (${kind}) proxy ownership transferred through proxy admin`);
-      } else {
-        console.log(FAILURE_CROSS + `${address} (${kind}) proxy ownership not affected by proxy admin`);
+    if (!opts.silent) {
+      const { provider } = hre.network;
+      const manifest = await Manifest.forNetwork(provider);
+      const { proxies } = await manifest.read();
+      const adminAddress = await admin.getAddress();
+
+      const affected = proxies.filter(async proxy => (await getAdminAddress(provider, proxy.address)) === adminAddress);
```

Comment:
Can you filter using an async function without some `await Promise.all` ?

Reply:
```suggestion
      const affected = await Promise.all(proxies.map(
          proxy => getAdminAddress(provider, proxy.address).then(controller => controller === adminAddress && proxy)
      )).then(array => array.filter(Boolean));
```

Reply:
I find this syntax quite confusing and have changed it to use a simple for-loop instead.